app/assets/javascripts/discourse/app/components/navigation-bar.gjs

```diff
@@ -0,0 +1,124 @@
+import Component from "@glimmer/component";
+import { concat, hash } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import { service } from "@ember/service";
+import DropdownMenu from "discourse/components/dropdown-menu";
+import NavigationItem from "discourse/components/navigation-item";
+import PluginOutlet from "discourse/components/plugin-outlet";
+import { filterTypeForMode } from "discourse/lib/filter-mode";
+import icon from "discourse-common/helpers/d-icon";
+import DMenu from "float-kit/components/d-menu";
+
+export default class NavigationBarComponent extends Component {
+  @service site;
+
+  get filterType() {
+    return filterTypeForMode(this.args.filterMode);
+  }
+
+  get selectedNavItem() {
+    const { navItems, connectors, category } = this.args;
```

Comment:
I know you just ported it as-is... but all this `connectors` logic seems very weird/outdated. The only place we use `<NavigationBar` in core is here:

https://github.com/discourse/discourse/blob/9641835743899e19c9a45f129d5c69c8e4d3b922/app/assets/javascripts/discourse/app/components/d-navigation.hbs#L21C3-L26C5

And `@connectors` is not passed.

So... I think we can just drop any logic which is looking at `args.connectors`. What do you think @chapoi @renato?

Reply:
I was **just** about to say the same thing, remarkable

Reply:
I found the culprit: https://github.com/discourse/discourse/commit/2bcb5360b602f1672bf73a848c22c6d42f08309d

So we can remove connectors here

```suggestion
    const { navItems, category } = this.args;
```

(and delete the block in my other comment)